third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -220,17 +227,30 @@ std::vector<ComputationClient::ComputationPtr> PjRtComputationClient::Compile(
     std::unique_ptr<xla::PjRtLoadedExecutable> executable =
         ConsumeValue(client_->Compile(instance.computation, compile_options));
 
-    const auto& hlo_modules = ConsumeValue(executable->GetHloModules());
-    HloComputation* hlo_computation = hlo_modules[0]->entry_computation();
-    xla::ProgramShape program_shape =
-        xla::ProgramShape(hlo_computation->ToProto().program_shape());
+    if (instance.is_sharded) {
```

Comment:
I guess this means CAPI won't support our spmd case yet?

Reply:
Right. Not until `GetHloModules` is implemented (which is WIP)